src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+        } catch (DataProcessingException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't add shopping cart " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("select sc from ShoppingCart sc "
+                            + "left join fetch sc.tickets t "
+                            + "left join fetch t.movieSession ms "
+                            + "left join fetch t.user "
+                            + "left join fetch ms.movie "
+                            + "left join fetch ms.cinemaHall "
+                            + "where sc.user.id = :userId", ShoppingCart.class)
```

Comment:
For the future, it is better to write all commands in upper case
```suggestion
            return session.createQuery("SELECT sc FROM ShoppingCart sc "
                            + "LEFT JOIN FETCH sc.tickets t "
                            + "LEFT JOIN FETCH t.movieSession ms "
                            + "LEFT JOIN FETCH t.user "
                            + "LEFT JOIN FETCH ms.movie "
                            + "LEFT JOIN FETCH ms.cinemaHall "
                            + "WHERE sc.user.id = :userId", ShoppingCart.class)
```